### PR TITLE
Stabilize loop test, fix 3306, disable unstable tests

### PR DESCRIPTION
**Pull request by gpotter2**
- fix https://github.com/secdev/scapy/issues/3306
- stabilize various networking test
- stabilize import tests
- fix BPF being loaded on linux during a mock test which caused `ioctl` to fail. Loading BPF on linux is now impossible

### CODE REVIEW

File: scapy/arch/linux.py [hunk 1]
```diff
@@ -600,7 +600,7 @@ def send(self, x):
         # type: (Packet) -> int
         iff = x.route()[0]
         if iff is None:
-            iff = conf.iface
+            iff = network_name(conf.iface)
```
Comment (mwutzke) [inline scapy/arch/linux.py:603]: I suspect L2Socket needs a similar change (line 502) for the case where `iface` is None (line 479)

Comment (gpotter2) [reply scapy/arch/linux.py:603]: No that's already taken care of line 479

File: scapy/layers/l2.py [hunk 2]
```diff
@@ -506,7 +506,10 @@ def mysummary(self):
 
 def l2_register_l3_arp(l2, l3):
     # type: (Type[Packet], Type[Packet]) -> Optional[str]
-    return getmacbyip(l3.pdst)
+    # TODO: support IPv6?
```
Comment (guedou) [inline scapy/layers/l2.py:509]: What is the goal of this change?
FTR, we do not need to support IPv6 here.

**Review by guedou — CHANGES_REQUESTED** [review]
Do we need to protect the other assignment of conf.iface?

File: scapy/layers/l2.py (hunk 2 above)
Comment (gpotter2) [reply scapy/layers/l2.py:509]: The comment or the changes to the function?
`ARP`'s pdst can be an IPv6 address in which case calling `getmacbyip` on it wouldn't make sense. I was unsure what we should do in that case

Comment (guedou) [reply scapy/layers/l2.py:509]: I did not know that ARP could carry IPv6 address. In that case, I might be useful to display a warning for everytheing else than IPv4 explaining that the address type is not supported.

Comment (gpotter2) [reply scapy/layers/l2.py:509]: Good idea. I've done that.
This is ready for review !

**Comment by gpotter2** [comment]
This PR should greatly improve stability of the tests.
Feel free to merge it @guedou

**Comment by gpotter2** [comment]
Self-merging because we need tests to pass and all the comments have been addressed :)

**Pull request merged**